lxd/storage/drivers/driver_btrfs_utils.go

```diff
@@ -390,3 +372,15 @@ func (d *btrfs) volumeSize(vol Volume) string {
 
 	return size
 }
+
+// setSubvolumeReadonlyProperty sets the readonly property on the subvolume to true or false.
+func (d *btrfs) setSubvolumeReadonlyProperty(path string, readonly bool) error {
+	// Silently ignore requests to set subvolume to readonly if running in a user namespace as it would mean
+	// we cannot undo this action.
+	if readonly && d.state.OS.RunningInUserNS {
```

Comment:
@stgraber can you check I've understood this correctly. Thanks

Reply:
We should just return nil if d.state.OS.RunningInUserNS is true as in the unlikely event that something is already read-only, you won't be able to set it writable either.

Reply:
@stgraber ok thanks will do.